### PR TITLE
Map changes to banner patterns in 22w18a

### PR DESCRIPTION
**Pull request by haykam821**

### CODE REVIEW

File: mappings/net/minecraft/block/entity/BannerPatterns.mapping [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_7446 net/minecraft/block/entity/BannerPatterns
+	METHOD method_43721 initAndGetDefault (Lnet/minecraft/class_2378;)Lnet/minecraft/class_2582;
```
Comment (liach) [inline mappings/net/minecraft/block/entity/BannerPatterns.mapping:2]:
```suggestion
	METHOD method_43721 registerAndGetDefault (Lnet/minecraft/class_2378;)Lnet/minecraft/class_2582;
```
See `MaterialRule` `MaterialCondition` `DensityFunctionTypes` `PaintingVariants`

Comment (haykam821) [reply mappings/net/minecraft/block/entity/BannerPatterns.mapping:2]: The `initAndGetDefault` method name is used for dimension types, structure sets, world presets, and flat level generator presets. While the inconsistency should be fixed, I think it is out of scope for this pull request.

Comment (liach) [reply mappings/net/minecraft/block/entity/BannerPatterns.mapping:2]: Hmm, turns out the `Registry` ones use `registerAndGetDefault` and `BuiltinRegistries` ones use `initAndGetDefault`

Comment (apple502j) [reply mappings/net/minecraft/block/entity/BannerPatterns.mapping:2]: Kinda makes sense given DRM registries are dynamic (although I believe `BuiltingRegistries` is not the ones they register to)

**Review by apple502j — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Pull request merged**